perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -490,4 +490,76 @@ public MemberGroupStatus getUnifiedMemberGroupStatus(PerunSession sess, User use
 
 	}
 
+	@Override
+	public List<Member> findMembers(PerunSession sess, Vo vo, String searchString, boolean onlySponsored) throws InternalErrorException {
+		Set<Member> members = new HashSet<>();
+
+		String voIdQueryString = "";
+		if(vo != null) {
+			voIdQueryString = " members.vo_id=" + vo.getId() + " and ";
+		}
+
+		String sponsoredQueryString = "";
+		if(onlySponsored) {
+			//only sponsored members
+			sponsoredQueryString+=" members.sponsored=1 and ";
+		}
+
+		String userNameQueryString = "";
+		if (Compatibility.isOracle()) {
+			userNameQueryString = " lower("+Compatibility.convertToAscii("u.first_name || u.middle_name || u.last_name")+") like '%' || ? || '%' ";
+		} else if (Compatibility.isPostgreSql()) {
+			userNameQueryString= " strpos(lower("+Compatibility.convertToAscii("COALESCE(u.first_name,'') || COALESCE(u.middle_name,'') || COALESCE(u.last_name,'')")+"),?) > 0 ";
+		} else if (Compatibility.isHSQLDB()) {
+			userNameQueryString=" lower("+Compatibility.convertToAscii("COALESCE(u.first_name,'') || COALESCE(u.middle_name,'') || COALESCE(u.last_name,'')")+") like '%' || ? || '%' ";
+		} else {
+			throw new InternalErrorException("Unsupported db type");
+		}
+
+		String idQueryString = "";
+		try {
+			int id = Integer.parseInt(searchString);
+			idQueryString = " members.user_id=" + id + " or members.id=" + id + " or ";
+		} catch (NumberFormatException e) {
+			// IGNORE wrong format of ID
+		}
+
+		String lowercaseSearchString = searchString.toLowerCase();
+
+		//searching by member mail
+		//searching by user preferredMail
+		//searching by login in userExtSources
+		//searching by login in logins (all namespaces)
+		//searching by name for user
+		//searching by user and member id
+		members.addAll(jdbc.query("select distinct " + memberMappingSelectQuery +
+				" from members " +
+				" left join users u on members.user_id=u.id " +
+				" left join member_attr_values mav1 on members.id=mav1.member_id and mav1.attr_id in (select id from attr_names where attr_name='urn:perun:member:attribute-def:def:mail') " +
+				" left join user_attr_values uav1 on u.id=uav1.user_id and uav1.attr_id in (select id from attr_names where attr_name='urn:perun:user:attribute-def:def:preferredMail') " +
+				" left join user_attr_values uav2 on u.id=uav2.user_id and uav2.attr_id in (select id from attr_names where friendly_name like 'login-namespace:%') " +
+				" left join user_ext_sources ues on ues.user_id=u.id " +
+				" where " +
+				voIdQueryString +
+				sponsoredQueryString +
+				" ( " +
+				" lower(mav1.attr_value)=lower(?) or " +
+				" lower(uav1.attr_value)=lower(?) or " +
+				" lower(uav2.attr_value)=lower(?) or " +
+				" lower(ues.login_ext)=lower(?) or " +
+				idQueryString +
+				userNameQueryString +
+				" ) ", MEMBER_MAPPER, lowercaseSearchString, lowercaseSearchString, lowercaseSearchString, lowercaseSearchString, lowercaseSearchString));
+
+		/**
```

Comment:
Is this Javadoc necessary? It could be at least written as a normal comment so there will be no warnings about dangling Javadoc.

Reply:
Removed.